### DISCUSSION TITLE
Feature Request: Bulk Actions, Export Device-Information, Return to Service (iOS)

### DISCUSSION THREAD

**Discussion by Adm61** [Micke-K/IntuneManagement#286] [Ideas]
Hey Mikael,

First of all, thank you so much for creating and maintaining the IntuneManagement tool. It’s an incredible resource, and I greatly appreciate the effort and dedication you’ve put into it. It has already proven to be immensely useful in managing multiple Intune instances.

I’d like to suggest a couple of features that I believe would add even more value to the tool:
	1.	Device Bulk Actions: e.g. Sync
	2.	CSV Export for Devices:
	•	The exported CSV could include details like device name, operating system, MAC-address, and—most importantly—the enrollment profile associated with each device.
	3.	Return to Service Feature for iOS Devices:
	•	Other MDM like Jamf and Relution offer a “Return to Service” feature for iOS devices, which allows administrators to prepare a device for the next user by automatically clearing certain settings or profiles and re-enrolling the device with a predefined WiFi-Profile.
	•	Would it be possible to implement a similar functionality in IntuneManagement Tool? This would be a game-changer for managing shared devices in education or enterprise environments.

Thank you so much!

All the best

**Comment by Micke-K** [comment]
Hello,,

So sorry. I completely missed the discussions the last month. I've been too busy, so I haven't had much time looking into things.

1. Possible but never thought about it. I might look into this at some point.
2. Also possible but you should be able to do this already by exporting devices. I haven't look what columns you get though. The enrolment profile is an Autopilot property. Are you are looking for a common report eg properties from Autopilot and Intune?
3. This might be a challenge. Not clue what this would be in Intune.  Are you looking for a solution for Company owned devices? Sounds something like a wipe and re-enroll with WiFi profile. I don't have a test environment with iOS devices so will be hard to develop something for it.

**Comment by Adm61** [comment]
Hi,

Thanks for your response! I completely understand that you're busy, and I really appreciate you taking the time to reply.

Regarding the points:

Device Bulk Actions: This remains the feature I miss the most. I did write a script to handle some of this functionality, but it would be far more convenient to perform these actions directly within the IntuneManagement Tool.

CSV Export for Devices: My main issue here is that when I use the export function under Devices in Intune, the enrollment profile isn’t included. If I export from the token section, I can see the enrollment profiles, but I don’t get hardware details like MAC addresses. A combined export showing both enrollment profiles and hardware information would be incredibly useful.

Return to Service Feature: This is more of a "nice to have" for me. I understand it’s a complex feature and would likely take a lot of effort to implement. It’s not urgent, as Microsoft might eventually release something similar themselves.

Thanks again for maintaining this tool—it’s been an invaluable resource!

All the best,
Adem
